packages/scoutgame-ui/src/components/accounts/components/NotificationSettings.tsx

```diff
@@ -0,0 +1,112 @@
+    handleSubmit,
+    reset,
+    formState: { isDirty }
+  } = useForm<UpdateUserNotificationSettingsFormValues>({
+    resolver: yupResolver(updateUserNotificationSettingsSchema),
+    mode: 'onChange',
+    defaultValues: {
+      emailNotification: user.sendTransactionEmails,
+      farcasterNotification: user.sendFarcasterNotification
+    }
+  });
+
+  const { execute, isExecuting } = useAction(updateUserNotificationSettingsAction, {
+    async onSuccess({ input }) {
+      setErrors(null);
+      reset(input);
+      revalidatePathAction();
+    },
+    onError(err) {
+      const hasValidationErrors = err.error.validationErrors?.fieldErrors;
+      const errorMessage = hasValidationErrors
+        ? concatenateStringValues(err.error.validationErrors!.fieldErrors)
+        : err.error.serverError?.message || (err.error.serverError as any)?.message || 'Something went wrong';
+
+      setErrors(errorMessage instanceof Array ? errorMessage : [errorMessage]);
+      log.error('Error saving email settings', { error: err });
+    }
+  });
+
+  const onSubmit = (data: UpdateUserNotificationSettingsFormValues) => {
+    execute(data);
+  };
+
+  function onInvalid(fieldErrors: FieldErrors) {
+    setErrors(['The form is invalid. Please check the fields and try again.']);
+    log.warn('Invalid form submission', { fieldErrors, values: getValues() });
+  }
+
+  return (
+    <Paper elevation={2} sx={{ p: 2 }}>
+      <form noValidate onSubmit={handleSubmit(onSubmit, onInvalid)}>
+        <Stack gap={2}>
+          <Stack direction='row' gap={1} alignItems='center'>
+            <NotificationsNoneOutlinedIcon />
+            <Typography variant='h6'>Notifications</Typography>
+          </Stack>
+          <Stack gap={{ xs: 1, md: 0 }}>
+            <Controller
+              control={control}
+              name='emailNotification'
+              render={({ field }) => (
+                <FormControlLabel
+                  disabled={isExecuting}
+                  control={<Checkbox checked={field.value} {...field} sx={{ alignSelf: 'flex-start' }} />}
+                  label='Enable Scout Game email notifications (activity and pending actions)'
+                />
+              )}
+            />
+            <Controller
+              control={control}
+              name='farcasterNotification'
+              render={({ field }) => (
+                <FormControlLabel
+                  control={<Checkbox checked={field.value} {...field} sx={{ alignSelf: 'flex-start' }} />}
+                  label='Enable Farcaster notifications'
+                />
+              )}
+            />
+          </Stack>
+          <LoadingButton
+            variant='contained'
+            color='primary'
+            type='submit'
+            loading={isExecuting}
+            disabled={isExecuting || !isDirty}
+            sx={{ width: 'fit-content' }}
+          >
+            Save
+          </LoadingButton>
+          <FormErrors errors={errors} />
```

Comment:
nice work getting the Formerrors and everything hooked up properly for this form.. i don't think i would have even been able to remember it all